internal/controller/postgrescluster/pgbackrest.go

```diff
@@ -873,6 +874,27 @@ func generateBackupJobSpecIntent(ctx context.Context, postgresCluster *v1beta1.P
 		// to read certificate files
 		jobSpec.Template.Spec.SecurityContext = postgres.PodSecurityContext(postgresCluster)
 		pgbackrest.AddConfigToCloudBackupJob(postgresCluster, &jobSpec.Template)
+
+		// If the user has specified a PVC to use as a log volume via the PGBackRestCloudLogVolume
+		// annotation, check for the PVC. If we find it, mount it to the backup job.
+		// Otherwise, create a warning event.
+		if logVolumeName := postgresCluster.Annotations[naming.PGBackRestCloudLogVolume]; logVolumeName != "" {
+			logVolume := &corev1.PersistentVolumeClaim{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      logVolumeName,
+					Namespace: postgresCluster.GetNamespace(),
+				},
+			}
+			err := errors.WithStack(r.Client.Get(ctx,
+				client.ObjectKeyFromObject(logVolume), logVolume))
+			if err != nil {
```

Comment:
Would we want to differentiate between a NotFound error vs. any other type of Error?

Reply:
Hmm... My thinking was that either way, the result was the same: we were unable to retrieve the PVC, so we should create a warning event. And the error message is passed into the event message, so the user will see what went wrong there.

Reply:
Doy, I misread. So, going back to basics:

a) if there's an annotation and the PVC exists, create the job with the PVC, and add it to the config
b) if there's an annotation and no PVC, create the job without the logging and issue an event about the missing logging volume -- backups > logging
c) if there's no annotation, create the job

And in that case, any error finding the PVC falls into the category of "backups are more important than logging" (and if there's another underlying problem, like the API is unreachable, we'll hear about it later, e.g., when we try to create the job).